owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt

```diff
@@ -300,5 +303,35 @@ fun Activity.showBiometricDialog(iEnableBiometrics: IEnableBiometrics) {
             dialog.dismiss()
         }
     }.show()
+
+}
+
+fun Activity.checkPasscodeDelayEnforced() {
+    val sharedPreferencesProvider = SharedPreferencesProviderImpl(this)
+
+    val lockEnforcedDelay = this.resources.getInteger(R.integer.lock_delay_enforced)
+    val passcodeConfigured = sharedPreferencesProvider.getBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, false)
+    val patternConfigured = sharedPreferencesProvider.getBoolean(PatternActivity.PREFERENCE_SET_PATTERN, false)
+
+    if (lockEnforcedDelay > LockTimeout.DISABLED.ordinal && (passcodeConfigured || patternConfigured)) {
+        when (lockEnforcedDelay) {
+            LockTimeout.IMMEDIATELY.ordinal -> {
```

Comment:
I wouldn't rely on ordinals, let's parse the integer to a LockTimeout and use the advantages of enum classes. We will have an exhaustive `when` if we do it that way. What do you think?

Reply:
Changed